src/components/structures/RoomSubListHeader.js

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var React = require('react');
+var ReactDOM = require('react-dom');
+var classNames = require('classnames');
+var sdk = require('matrix-react-sdk')
+var FormattingUtils = require('matrix-react-sdk/lib/utils/FormattingUtils');
+var RoomNotifs = require('matrix-react-sdk/lib/RoomNotifs');
+var AccessibleButton = require('matrix-react-sdk/lib/components/views/elements/AccessibleButton');
+var ConstantTimeDispatcher = require('matrix-react-sdk/lib/ConstantTimeDispatcher');
+
+module.exports = React.createClass({
+    displayName: 'RoomSubListHeader',
+
+    propTypes: {
+        label: React.PropTypes.string.isRequired,
+        tagName: React.PropTypes.string,
+        roomCount: React.PropTypes.oneOfType([
+            React.PropTypes.string,
+            React.PropTypes.number
+        ]),
+        collapsed: React.PropTypes.bool.isRequired, // is LeftPanel collapsed?
+        isIncomingCallRoom: React.PropTypes.bool,
+        roomNotificationCount: React.PropTypes.array,
+        hidden: React.PropTypes.bool,
+        onClick: React.PropTypes.func,
+        onHeaderClick: React.PropTypes.func,
+    },
+
+    getDefaultProps: function() {
+        return {
+            onHeaderClick: function() {}, // NOP
+        };
+    },
+
+    componentWillMount: function() {
+        // constantTimeDispatcher.register("RoomSubList.refreshHeader", this.props.tagName, this.onRefresh);
+    },
```

Comment:
i left them in just in case we move the refreshHeader back to the child in the end - it got moved back to RoomSubList because we also have header info in the truncation ellipsis cells.  agreed it's bad to leave commented out stuff hanging around though.